src/transports/ssh.c

```diff
@@ -893,17 +978,101 @@ int git_transport_ssh_with_paths(git_transport **out, git_remote *owner, void *p
 #endif
 }
 
+static const char *supported_filenames[] = {
+#ifdef __APPLE__
+	"libssh2.1.dylib",
+#else
+	"libssh2.so.1",
+#endif
+	NULL,
+};
+
+/**
+ * Try to load a supported version of libssh2
+ */
+static void load_libssh2(void)
+{
+#ifdef GIT_WIN32
+	/* Windows doesn't do SOVERSIONs but you ship your own library so we're probably OK */
+	git_libssh2_handle = LoadLibrary("libssh2.dll");
```

Comment:
some environments (msys2 probably, will have to check) approximate sonames via libssh2-#.dll